assets/app/scripts/directives/deleteLink.js

```diff
@@ -1,21 +1,34 @@
 'use strict';
 
 angular.module("openshiftConsole")
-  .directive("deleteLink", function ($uibModal, $location, $filter, hashSizeFilter, APIService, DataService, AlertMessageService, Navigate, Logger) {
+  .directive("deleteLink", function ($uibModal, $location, $filter, $q, hashSizeFilter, APIService, DataService, AlertMessageService, Navigate, Logger) {
     return {
       restrict: "E",
       scope: {
+        // Resource Kind to delete (e.g., "Pod" or "ReplicationController").
         kind: "@",
+        // Optional resource group.
+        group: "@?",
         // Optional display name for kind.
         typeDisplayName: "@?",
+        // Name of the resource to delete.
         resourceName: "@",
+        // The name of the resource's project. Optional if kind === "Project".
         projectName: "@",
+        // Alerts object for success and error alerts.
         alerts: "=",
+        // Optional display name of the resource to delete.
         displayName: "@",
         // Set to true to disable the delete button.
         disableDelete: "=?",
-        // Only show the button and no text.
-        buttonOnly: "@"
+        // Optional link label. Defaults to "Delete".
+        label: "@?",
+        // Only show a delete icon with no text.
+        buttonOnly: "@",
+        // Stay on the current page without redirecting to the resource list.
+        stayOnCurrentPage: "=?",
+        // Array of associated HPAs for this resource. If set, prompts the user to delete the HPA resources as well.
+        hpaList: "=?"
```

Comment:
Yeah, I considered it. Right now, the dialog gives some specific help text about HPAs that might be hard to generalize. Thoughts?

https://github.com/jwforres/origin/pull/24/files/df39d33b4c94677b6b0dbbabddca69acfd858492#diff-833d750ce875560434dd51f7d42c3735R13